app/assets/javascripts/discourse/app/widgets/header.js

```diff
@@ -496,6 +504,14 @@ export default createWidget("header", {
   buildKey: () => `header`,
   services: ["router", "search"],
 
+  init() {
+    registerWidgetShim(
+      "extra-button",
+      "div.wrapper",
+      hbs`<LegacyHeaderButtonShim @component={{@data.component}} />`
```

Comment:
This shim component seems kinda unnecessary? Seems like this would be exactly identical?

```suggestion
      hbs`<@data.component}} />`
```